adsmp/app.py

```diff
@@ -345,17 +298,18 @@ def reindex(self, solr_docs, solr_urls, commit=False):
                         # here if body not in error message do not retry, just note as a fail
                         self.logger.error('Failed posting individual bibcode %s to Solr\nurls: %s, offending payload %s, error is %s', failed_bibcode, solr_urls, doc, e)
                         failed_bibcodes.append(failed_bibcode)
+        # next update postgres record
+        self.index_complete(solr_docs, failed_bibcodes, 'solr')
         return failed_bibcodes
 
-    
-    def mark_processed(self, bibcodes, type=None, status = None):
+    def mark_processed(self, bibcodes, type=None, status=None, update_timestamps=True):
         """Updates the timesstamp for all documents that match the bibcodes.
         Optionally also sets the status (which says what actually happened 
         with the document).
         
         """
 
-        if not self.update_timestamps:
+        if not self.update_timestamps or not update_timestamps:
```

Comment:
I do not have the impression we are using `self.update_timestamps` in practice. The value of that attribute comes from the env variable `UPDATE_TIMESTAMPS` that, as far as I know, we do not usually change. Should we remove it from the logic to make the code more readable?

Reply:
Roman very recently added this feature so he can update test solr instances from the prod ingest pipeline and not modify the timestamps.  I think we need to keep the feature:
https://github.com/adsabs/ADSMasterPipeline/commit/f4864e7fc4da366b0032389bac1f0870ca97b764#diff-391108dd6764c0740978fdfcfb30ec4fe9b011425a846c11ec4092c1708914ef

Reply:
Thanks for that reminder! Then, is there any way we could merge this two parameters? I think having an objects' property and method's argument that have the same effect makes it a bit confusing. And actually, I do not see when we use it. In the config files or in our deployment files, we never set `UPDATE_TIMESTAMPS` so it looks like in practice it always has the default value which is `True`.